subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformedFileCache.java

```diff
@@ -161,6 +180,29 @@ private CacheKey getCacheKey(File inputFile, HashCode inputsHash) {
         return new CacheKey(inputFileSnapshot, inputsHash);
     }
 
+    @SuppressWarnings("Since15")
+    private void markAccessedByTouchingTopmostOutputDirectory(List<File> files) {
```

Comment:
⭕️ I guess the performance tests will tell us whether this is going to be a problem :)